### ISSUE TITLE
[⚡ Feature]: TurboRepo + NextJS deployed to Cloudflare Pages

### ISSUE THREAD

**Issue by rares-lupascu** [cloudflare/next-on-pages#415]
### Description

Hey

Is there a way to deploy a nextjs app within a monorepo to Cloudflare Pages?
I tried multiple build settings but i cannot get it to work?

Should i give up or these i a possibility for this tot work?

### Would you like to help?

- [ ] Would you like to help implement this feature?

**Comment by james-elicx** [comment]
Hey there,

Have you tried using a script in a package.json file to start building your application? For instance, in the Next.js app in your monorepo, you could have a script called `pages:build` and it could run `npx @cloudflare/next-on-pages`. Then, your build command on Cloudflare Pages could be something similar to `pnpm --filter @apps/web pages:build`, running the script in that app in your monorepo.

This strategy should build your application in the correct location. You would just need to update your output directory in your Pages project's build settings to reflect the location where your app is, e.g. `/apps/web/.vercel/output/static`.

**Comment by james-elicx** [comment]
> @james-elicx thank you ... it worked after a few minor fixes in settings
> 
> unfortunately:
> 
> ```
> 14:20:08.450 | ✘ [ERROR] Error: Pages only supports files up to 26.2 MB in size
> -- | --
> 14:20:08.450 |  
> 14:20:08.451 | cache/webpack/edge-server-production/0.pack.gz is 461 MB in size
> ```
> 
> too bad

Hello, it sounds like you might be trying to upload your `.next` folder contents.

Your output directory should be `{path to app}/.vercel/output/static` and it shouldn't be referring to `.next` at all, or trying to upload `.next/cache`.

**Comment by rares-lupascu** [comment]
@james-elicx u were again right ... but now

```
15:05:28.588 | Success: Assets published!
-- | --
15:05:30.872 | Error: Failed to publish your Function. Got error: Your Functions script is over the 1 MiB size limit (workers.api.error.script_too_large)
```

**Comment by rares-lupascu** [comment]
my problem is that i do not have a _worker.js file ... maybe because the build failed

**Comment by rares-lupascu** [comment]
are these the ones?
![image](https://github.com/cloudflare/next-on-pages/assets/3641866/6bd4d937-99b4-4cd7-bc87-6c62ca3e65cf)

**Comment by rares-lupascu** [comment]
just deployed this to vercel and checkout all the edge functions .... none of them is above 1MB but cumulated the size for sure is above 5MB

**Comment by james-elicx** [comment]
> are these the ones? ![image](https://user-images.githubusercontent.com/3641866/257826363-6bd4d937-99b4-4cd7-bc87-6c62ca3e65cf.png)

No, those are assets, not the `_worker.js` directory. You have to build it locally if you want to zip it and calculate the size of the worker directory to understand how much above your limit it is.

> just deployed this to vercel and checkout all the edge functions .... none of them is above 1MB but cumulated the size for sure is above 5MB

What Vercel deploys is very different to what we deploy. We try to minify functions as much as we can and also combine them into one main worker. The size Vercel tells you is not really related to the size we deploy to Pages.

**Comment by rares-lupascu** [comment]
damn ... it's 1.3 MB :))

**Comment by james-elicx** [comment]
> damn ... it's 1.3 MB :))

Sounds like you'll need Workers/Pages Paid then I'm afraid, since that increases the size limit from 1mb.

**Comment by rares-lupascu** [comment]
yeap ... thanks @james-elicx really appreciate it

**Comment by AlecKriebel** [comment]
> > are these the ones? ![image](https://user-images.githubusercontent.com/3641866/257826363-6bd4d937-99b4-4cd7-bc87-6c62ca3e65cf.png)
> 
> No, those are assets, not the `_worker.js` directory. You have to build it locally if you want to zip it and calculate the size of the worker directory to understand how much above your limit it is.
> 
> > just deployed this to vercel and checkout all the edge functions .... none of them is above 1MB but cumulated the size for sure is above 5MB
> 
> What Vercel deploys is very different to what we deploy. We try to minify functions as much as we can and also combine them into one main worker. The size Vercel tells you is not really related to the size we deploy to Pages.

@james-elicx are you specifically talking about the `_worker.js` directory in this comment? I'm trying to figure out exactly how to test bundle sizes locally. 

It sounds like what Cloudflare validates sizes against is a zipped version of the `/.vercel/output/static/_worker.js` directory, is that correct?

**Comment by james-elicx** [comment]
> @james-elicx are you specifically talking about the `_worker.js` directory in this comment? I'm trying to figure out exactly how to test bundle sizes locally.
> 
> It sounds like what Cloudflare validates sizes against is a zipped version of the `/.vercel/output/static/_worker.js` directory, is that correct?

If you zip the `_worker.js` directory, you should be able to get an idea of the zipped bundle size that is uploaded to Cloudflare, yes.

**Comment by pantharshit007** [comment]
hey @james-elicx  do you know anyway on how I can deploy my hono app from my turborepo
